examples/parallel_profiling.py

```diff
@@ -0,0 +1,142 @@
+"""
+================================================
+Memory and CPU consumption using joblib.Parallel
+================================================
+
+This example illustrates some important aspects to keep in mind when using
+:class:`joblib.Parallel`. We will focus on different patterns resulting to the
+same outcomes but which trigger different internal behaviors. Those differences
+will impact the memory consumption and computation time.
+
+To illustrate those differences, the moving average from a large array will be
+computed. Therefore, the average can be computed in parallel at different
+location of the signal.
+
+"""
+from __future__ import division
+
+print(__doc__)
+
+###############################################################################
+# `memory_used` and `print_info` are two helper functions being used
+# extensively later on. The former function allows to track the memory
+# consumption during the call of a function while the latter output profiling
+# information.
+
+import gc
+from memory_profiler import memory_usage
+
+
+def memory_used(func, *args, **kwargs):
```

Comment:
I think that I would integrate in this function the fact that you run things 10 times and that you return the mean; in order to make the code below lighter (and I would create a docstring, as a consequence).